pkg/admission/server.go

```diff
@@ -0,0 +1,142 @@
+package admission
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	admission "k8s.io/api/admission/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"net/http"
+	"path/filepath"
+	sav1 "sigs.k8s.io/service-apis/apis/v1alpha1"
+	"k8s.io/klog/v2"
+)
+
+const (
+	tlsDir      = `/run/secrets/tls`
+	tlsCertFile = `tls.crt`
+	tlsKeyFile  = `tls.key`
+)
+
+var (
+	scheme = runtime.NewScheme()
+	codecs = serializer.NewCodecFactory(scheme)
+)
+
+
+// ServeHTTP parses AdmissionReview requests and responds back
+// with the validation result of the entity.
+func ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if r.Body == nil {
+		// log Error("received request with empty body")
+		http.Error(w, "admission review object is missing",
+			http.StatusBadRequest)
+		return
+	}
+	data, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		klog.Errorf("failed to read request from client: %v\n", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
```

Comment:
Tiny nit: It looks like variations of these 3 lines are repeated a few times, maybe some kind of helper function could help.

Reply:
Yes I see that now, the helper is a good call